pkg/protocols/common/protocolstate/state.go

```diff
@@ -15,16 +16,34 @@ import (
 	"github.com/projectdiscovery/networkpolicy"
 	"github.com/projectdiscovery/nuclei/v3/pkg/types"
 	"github.com/projectdiscovery/nuclei/v3/pkg/utils/expand"
+	"github.com/projectdiscovery/utils/memguardian"
 )
 
 // Dialer is a shared fastdialer instance for host DNS resolution
-var Dialer *fastdialer.Dialer
+var (
+	Dialer      *fastdialer.Dialer
+	MemGuardian *memguardian.MemGuardian
+)
 
 // Init creates the Dialer instance based on user configuration
 func Init(options *types.Options) error {
 	if Dialer != nil {
 		return nil
 	}
+	if MemGuardian == nil {
+		mg, err := memguardian.New(
+			memguardian.WitInterval(time.Second),
+			memguardian.WithRatioWarning(75),
```

Comment:
can the ratio be configurable as a setting?

Reply:
same as my previous comment